lightning/src/ln/channelmanager.rs

```diff
@@ -3249,6 +3250,18 @@ where
 				if let Err((err, code)) = chan.htlc_satisfies_config(&msg, outgoing_amt_msat, outgoing_cltv_value) {
 					break Some((err, code, chan_update_opt));
 				}
+				match (msg.amount_rgb, outgoing_amount_rgb) {
+					(Some(_), None) | (None, Some(_)) => {
+						break Some(("Refusing to forward a non-authorized swap.", 0x1000 | 11, chan_update_opt));
+					},
+					(None, None) => {},
+					(Some(x), Some(y)) if x == y => {
+						log_trace!(self.logger, "Forwarding RGB payment of value: {}", x);
+					},
+					_ => {
+						break Some(("Refusing to forward a payment with non-matching RGB amount", 0x1000 | 11, chan_update_opt)); // amount_below_minimum
```

Comment:
not sure why there is a comment saying `amount_below_minimum`, it seems this match includes all cases where the amount is different from the requested one

Reply:
`amount_below_minimum` is the technical name of the error code we are returning here (code 11). See [bolt 4](https://github.com/lightning/bolts/blob/master/04-onion-routing.md#failure-messages)

I think that's the closest error code we can return, as there's no "above too high" error, or anything like that. Also in practice I assume other nodes won't generally try to gift assets to you like this (and if they really want to, they can just send a keysend payment). So in practice most of the times I assume we will be rejecting payments that try to scam the node by only sending `x` units of an assets and asking to forward `y`, with `y > x`

Reply:
I agree there are no better errors. What confused me is that the first match arm is returning the same error but has no comment. I propose to do the following:
```rust
(Some(_), None) | (None, Some(_)) => {
    // using amount_below_minimum error because there is no better error available
    break Some(("Refusing to forward a non-authorized swap.", 0x1000 | 11, chan_update_opt));
},
[...]
_ => {
    // using amount_below_minimum error because there is no better error available
    break Some(("Refusing to forward a payment with non-matching RGB amount", 0x1000 | 11, chan_update_opt));
}
```